src/main/java/com/zombiecastlerush/building/Combat.java

```diff
@@ -0,0 +1,44 @@
+package com.zombiecastlerush.building;
+
+import com.zombiecastlerush.entity.Role;
+import com.zombiecastlerush.util.Parser;
+import com.zombiecastlerush.util.Prompter;
+
+import java.util.List;
+import java.util.Random;
+
+public class Combat extends Challenge {
+    public Combat(String description) {
+        super(description);
+    }
+
+    public static void combat(Role player, Role enemy) {
+        playerFight(player, enemy);
```

Comment:
I think what would make much more sense here is to call `playerFight`, `enemyFight`, and evaluate the enemy/player statistics from there, rather than put all of that into `playerFight`. From a readability standpoint, it makes more sense, I think... and also encapsulation wise, `playerFight` is doing too much, while `combat` is doing too little. If `combat` only calls `playerFight`, then there's no need for you to have the `combat` method. On the other hand, having everything called from `playerFight` makes your method do too many things...

Best solution in my opinion is to have `combat` call `playerFight` and `enemyFight`, with the requisite condition checks (getHealth() < 0 etc)

Reply:
implemented and pushed recommended changes.